File: src/main/java/com/mbayou/kick4k/events/EventSubscriptionResponse.java
package com.mbayou.kick4k.events;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;

public class EventSubscriptionResponse {
    private final String error;
    private final String name;
    private final String subscriptionId;
    private final Integer version;

    @JsonCreator
    public EventSubscriptionResponse(@JsonProperty("error") String error,
                                     @JsonProperty("name") String name,
                                     @JsonProperty("subscription_id") String subscriptionId,
                                     @JsonProperty("version") Integer version) {
        this.error = error;
        this.name = name;
        this.subscriptionId = subscriptionId;
        this.version = version;
    }

    public String getError() {
        return error;
    }

    public String getName() {
        return name;
    }

    public String getSubscriptionId() {
        return subscriptionId;
    }

    public Integer getVersion() {
        return version;
    }
}

File: src/main/kotlin/com/mbayou/kick4k/events/EventSubscriptionResponse.kt
package com.mbayou.kick4k.events

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty

data class EventSubscriptionResponse @JsonCreator constructor(
    @JsonProperty("error") val error: String?,
    @JsonProperty("name") val name: String?,
    @JsonProperty("subscription_id") val subscriptionId: String?,
    @JsonProperty("version") val version: Int?,
)
